parlai/core/torch_classifier_agent.py

```diff
@@ -217,15 +220,31 @@ def _update_confusion_matrix(self, batch, predictions):
         """
         Update the confusion matrix given the batch and predictions.
 
-        :param batch:
-            a Batch object (defined in torch_agent.py)
         :param predictions:
             (list of string of length batchsize) label predicted by the
             classifier
+        :param batch:
+            a Batch object (defined in torch_agent.py)
         """
-        for i, pred in enumerate(predictions):
-            label = batch.labels[i]
-            self.metrics['confusion_matrix'][(label, pred)] += 1
+        f1_dict = {}
+        for class_name in self.class_list:
+            if not self.opt['get_all_metrics']:
+                class_name = self.ref_class
+            prec_str = f'class_{class_name}_prec'
+            recall_str = f'class_{class_name}_recall'
+            f1_str = f'class_{class_name}_f1'
+            precision, recall, f1 = ClassificationMetric.compute_metrics(
+                predictions, batch.labels, class_name
+            )
+            f1_dict[class_name] = f1
+            if not self.opt['get_all_metrics']:
```

Comment:
why are the metrics not being recorded at all `if not self.opt['get_all_metrics']`? this continue doesn't belong here